gdal_nodes.cpp

```diff
@@ -303,51 +296,86 @@ void OGRWriterNode::process()
     wkbType = wkbMultiPolygon25D;
   }
 
-  //  // Parse Layer Creation Options
-  //  std::vector<std::string> lco_vec;
-  //  std::stringstream        s_stream(lco);
-  //  while (s_stream.good()) {
-  //    std::string substr;
-  //    getline(s_stream, substr, ',');
-  //    lco_vec.push_back(substr);
-  //  }
-  //  char** papszOptionsLayer = nullptr;
-  //  papszOptionsLayer =
-  //    CSLSetNameValue(papszOptionsLayer, "APPEND_SUBDATASET", "YES");
-  //  for (auto & i : lco_vec) {
-  ////    papszOptionsLayer[i] = const_cast<char*>(lco_vec[i].c_str());
-  //  }
-
-  // TODO: appending features probably requires not creating a new layer, but instead getting the layer from the dataset with GetLayer()
-  // overwrite existing layers
   char** lco = nullptr;
-  lco = CSLSetNameValue(lco, "OVERWRITE", "YES");
-  poLayer = poDS->CreateLayer(manager.substitute_globals(layername).c_str(), &oSRS, wkbType, lco);
-  if (poLayer == nullptr) {
-    printf("Layer creation failed.\n");
-    exit(1);
+  if (overwrite_dataset)
+    lco = CSLSetNameValue(lco, "OVERWRITE", "YES");
+  else
+    lco = CSLSetNameValue(lco, "OVERWRITE", "NO");
+
+  int fcnt(0);
+  int layer_cnt = poDS->GetLayerCount();
+  if (layer_cnt > 0) {
+    poLayer = poDS->GetLayer(0);
+    // When the driver is PostgreSQL AND the tables= option is used in the
+    // connection string, the GetLayer() always gets a layer, even
+    // if the table doesn't exist. So we need to check for existing fields.
+    fcnt = poLayer->GetLayerDefn()->GetFieldCount();
+    if (fcnt == 0) {
+      append             = false;
+      overwrite_dataset  = false;
+      bool tables_in_dsn = manager.substitute_globals(filepath).find(
+                             "tables=") != std::string::npos;
+      if (tables_in_dsn) {
+        printf("You are creating a new table in PostgreSQL, but also specified "
+               "the 'tables=' option in the connection string. GDAL will throw "
+               "and error, the table name will be %s in the public schema, "
+               "unless you also passed the schemas= option.\n",
+               manager.substitute_globals(layername).c_str());
+      }
+    }
+  } else {
+    append            = false;
+    overwrite_dataset = false;
   }
 
-  // Create GDAL feature attributes
   std::unordered_map<std::string, size_t> attr_id_map;
-  int fcnt = poLayer->GetLayerDefn()->GetFieldCount();
-  for (auto& term : poly_input("attributes").sub_terminals()) {
-    std::string name = term->get_name();
-    //see if we need to map this name to another one
-    auto search = output_attribute_names.find(name);
-    if(search != output_attribute_names.end()) {
-      if(search->second.size()!=0)
-        name = search->second;
+  if (not append) {
```

Comment:
For portability better to use `!append` instead of `not append`. Visual Studio does not like `not`, `and`, `or` etc.

Reply:
Ok, noted!